pkg/jobs/worker.go

```diff
@@ -391,9 +402,14 @@ func (t *task) exec(ctx *WorkerContext) (err error) {
 func (t *task) nextDelay(prevError error) (bool, time.Duration, time.Duration) {
 	// for certain kinds of errors, we do not have a retry since these error
 	// cannot be recovered from
-	switch prevError {
-	case ErrAbort, ErrMessageUnmarshal, ErrMessageNil:
-		return false, 0, 0
+	{
+		if _, ok := prevError.(ErrBadTrigger); ok {
+			return false, 0, 0
+		}
+		switch prevError {
+		case ErrAbort, ErrMessageUnmarshal, ErrMessageNil:
+			return false, 0, 0
+		}
```

Comment:
Why not just this?

```go
switch prevError {
case ErrAbort, ErrMessageUnmarshal, ErrMessageNil, ErrBadTrigger:
        return false, 0, 0
}
```

Reply:
`ErrBadTrigger` is an error *type*, not a value.